docs/extending-and-experimenting.md

```diff
@@ -149,6 +149,171 @@ of the unsubscription (`teardown`) and update
 some state. This all can be reapplied when you write your
 own APIs.
 
+## Creating your first exchange
+
+### Authentication
+
+Managing and refreshing tokens is a very common case in
+modern application development. In this part we'll build
+this exchange from scratch.
+
+> All wonka operators used can be found [here](https://wonka.kitten.sh/api/)
+
+So let's start with the basic template for an exchange
+
+```js
+import { pipe } from 'wonka';
+
+export const refreshTokenExchange = ({ forward }) => {
+  return operations$ => {
+    return pipe(
+      operations$,
+      forward
+    );
+  };
+};
+```
+
+As of now it enters the exchange and tells it to continue due
+to forward being invoked. So this is basically an empty
+exchange.
+
+Next up is writing some code that refreshes our token, so imagine
+the following method:
+
+> note that these methods should be replaced with your own logic.
+
+```js
+const query = gql`
```

Comment:
I think we should show this example with just REST, or even with imaginary `fetchToken` functions. It's a minor detail, since people know how to write those and if we want to forward a "refresh token" query there's also an option of forwarding it to the next exchange, which is a little trickier, so not worth doing here imo